plugin/pkg/scheduler/core/generic_scheduler.go

```diff
@@ -127,7 +127,20 @@ func (g *genericScheduler) Schedule(pod *v1.Pod, nodeLister algorithm.NodeLister
 	}
 
 	trace.Step("Selecting host")
-	return g.selectHost(priorityList)
+	host, err := g.selectHost(priorityList)
+	if err != nil {
+		return "", err
+	}
+
+	if len(g.extenders) != 0 {
+		for _, extender := range g.extenders {
+			err := extender.Bind(pod, host)
```

Comment:
@ravigadde , it not about kubelet start. In kube-mesos refactor, I try to avoid cache which makes it complex. And I prefer no interaction (e.g. recover/re-schedule) between scheduler & extender; predicates/prefer are standalone, no interaction with other part, e.g. apiserver bind failed.